pallets/grants/src/migrations.rs

```diff
@@ -0,0 +1,165 @@
+
+			// Write to the new storage with removed and added fields
+			for (account, old_vesting) in stored_data {
+				translated.saturating_inc();
+				max_schedules = max_schedules.max(old_vesting.len());
+				if let Ok(new_vesting) = BoundedVec::<VestingScheduleOf<T>, T::MaxSchedule>::try_from(old_vesting) {
+					<VestingSchedules<T>>::insert(account, new_vesting);
+				} else {
+					log::error!(
+						"on_runtime_upgrade, Could be boundedvec Overflow at row index [{:#?}]",
+						translated
+					);
+				}
+			}
+
+			<StorageVersion<T>>::put(crate::Releases::V1);
+
+			log::info!(
+				"on_runtime_upgrade[{:#?}]=> Upgraded {} schedules, Max {}, storage to version {:?}",
+				line!(),
+				translated,
+				max_schedules,
+				<StorageVersion<T>>::get()
+			);
+
+			T::DbWeight::get().reads_writes(translated + 1, translated + 1)
+		} else {
+			log::info!(
+				"on_runtime_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+				line!(),
+			);
+			T::DbWeight::get().reads(1)
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	pub fn pre_upgrade<T: Config>() -> Result<(), &'static str> {
+		log::info!(
+			"pre_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+			line!(),
+			crate::Releases::V1,
+			<StorageVersion<T>>::get(),
+		);
+
+		if <StorageVersion<T>>::get() == Releases::V0 {
+			let pallet_prefix: &[u8] = b"Vesting";
+			let storage_item_prefix: &[u8] = b"VestingSchedules";
+
+			// Check number of entries, and set it aside in temp storage
+			let stored_data: Vec<_> = storage_key_iter::<T::AccountId, Vec<VestingScheduleOf<T>>, Blake2_128Concat>(
+				pallet_prefix,
+				storage_item_prefix,
+			)
+			.collect();
+
+			stored_data
+				.iter()
+				.try_for_each(|(_account, old_vesting)| -> Result<(), &'static str> {
+					if old_vesting.len() > T::MaxSchedule::get() as usize {
+						Err("Vesting length is above MaxSchedule")?;
+					}
+					Ok(())
+				})?;
+
+			log::info!(
+				"pre_upgrade[{:#?}]=> VestingSchedules map count :: [{:#?}]",
+				line!(),
+				stored_data.len(),
+			);
+		} else {
+			log::info!(
+				"pre_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+				line!(),
+			);
+		}
+		Ok(())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	pub fn post_upgrade<T: Config>() -> Result<(), &'static str> {
+		log::info!(
+			"post_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+			line!(),
+			crate::Releases::V1,
+			<StorageVersion<T>>::get(),
+		);
+
+		if <StorageVersion<T>>::get() == Releases::V1 {
+			let pallet_prefix: &[u8] = b"Vesting";
+			let storage_item_prefix: &[u8] = b"VestingSchedules";
+
+			let mapping_count: u32 = storage_key_iter::<
+				T::AccountId,
+				BoundedVec<VestingScheduleOf<T>, T::MaxSchedule>,
+				Blake2_128Concat,
+			>(pallet_prefix, storage_item_prefix)
+			.count() as u32;
+
+			log::info!(
+				"post_upgrade[{:#?}]=> VestingSchedules map count :: [{:#?}]",
+				line!(),
+				mapping_count,
+			);
```

Comment:
We could have a better test for post upgrade to actually return Err for some unexpected cases, however it could be an improvement and I will not hold this PR for that. Especially that we have a good test pre-upgrade.